module/lua/llimits.h

```diff
@@ -122,6 +122,12 @@ typedef LUAI_UACNUMBER l_uacNumber;
 #define LUAI_MAXCCALLS		20
 #endif
 
+/*
+ * Minimum amount of available stack space (in bytes) to make a C call.  With
+ * gsub() recursion, the stack space between each luaD_call() is 1256 bytes.
+ */
+#define LUAI_MINCSTACK		4096
```

Comment:
Can we safely decrease this further for 32-bit platforms where the default on x86_32 is still 8K.

Reply:
Probably, but 4K might also work fine (leaving 4K for normal LUA stuff). Do you know how to test this?  It looks like there's a "Ubuntu 16.04 i386" build, but not an associated test?

Reply:
There a 32-bit builder to test the compile but none that run the test suite.  We'd need to manually test in a VM.  4K may be fine, though leaving 4K for everything else feels a bit tight to me.  Using an Ubuntu 16.04 iso is probably the easiest route.